File: app/src/main/java/net/cynreub/subly/domain/repository/PaymentMethodRepository.kt
package net.cynreub.subly.domain.repository

import kotlinx.coroutines.flow.Flow
import net.cynreub.subly.domain.model.PaymentMethod
import java.util.UUID

interface PaymentMethodRepository {
    fun getAllPaymentMethods(): Flow<List<PaymentMethod>>
    fun getPaymentMethodById(id: UUID): Flow<PaymentMethod?>
    suspend fun insertPaymentMethod(paymentMethod: PaymentMethod)
    suspend fun updatePaymentMethod(paymentMethod: PaymentMethod)
    suspend fun deletePaymentMethod(paymentMethod: PaymentMethod)
    suspend fun deletePaymentMethodById(id: UUID)
    suspend fun getSubscriptionCountForPaymentMethod(paymentMethodId: UUID): Int
}

File: subly_ios/Subly/Subly/Domain/Repositories/PaymentMethodRepository.swift
import Foundation
import Combine

protocol PaymentMethodRepository {
    func getAllPaymentMethods() -> AnyPublisher<[PaymentMethod], Never>
    func getPaymentMethodById(_ id: UUID) -> AnyPublisher<PaymentMethod?, Never>
    func insertPaymentMethod(_ paymentMethod: PaymentMethod) async throws
    func updatePaymentMethod(_ paymentMethod: PaymentMethod) async throws
    func deletePaymentMethod(_ paymentMethod: PaymentMethod) async throws
    func deletePaymentMethod(id: UUID) async throws
    func getSubscriptionCount(for paymentMethodId: UUID) async -> Int
}
